Ignore "Factory" which is not available with system tzdata

Backport various fix from 5.6

--- a/ext/date/tests/bug66985.phpt	2014-10-30 07:32:03.297693403 +0100
+++ b/ext/date/tests/bug66985.phpt	2014-10-30 07:32:45.138877977 +0100
@@ -3,7 +3,7 @@
 --FILE--
 <?php
 $zones = array(
-	"CST6CDT", "Cuba", "Egypt", "Eire", "EST5EDT", "Factory", "GB-Eire",
+	"CST6CDT", "Cuba", "Egypt", "Eire", "EST5EDT", "GB-Eire",
 	"GMT0", "Greenwich", "Hongkong", "Iceland", "Iran", "Israel", "Jamaica",
 	"Japan", "Kwajalein", "Libya", "MST7MDT", "Navajo", "NZ-CHAT", "Poland",
 	"Portugal", "PST8PDT", "Singapore", "Turkey", "Universal", "W-SU",
@@ -45,11 +45,6 @@
 )
 DateTimeZone Object
 (
-    [timezone_type] => 3
-    [timezone] => Factory
-)
-DateTimeZone Object
-(
     [timezone_type] => 3
     [timezone] => GB-Eire
 )
From 7cbd8f8b8a662c1e74cb1c97dac6f606f8f477c9 Mon Sep 17 00:00:00 2001
From: krakjoe <joe.watkins@live.co.uk>
Date: Thu, 21 Aug 2014 10:03:48 +0100
Subject: [PATCH] remove bogus locale use from test

---
 ext/standard/tests/strings/setlocale_variation2.phpt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ext/standard/tests/strings/setlocale_variation2.phpt b/ext/standard/tests/strings/setlocale_variation2.phpt
index 038ba58..10ae22f 100644
--- a/ext/standard/tests/strings/setlocale_variation2.phpt
+++ b/ext/standard/tests/strings/setlocale_variation2.phpt
@@ -18,8 +18,11 @@ if (substr(PHP_OS, 0, 3) == 'WIN') {
 /* setlocale() to set all available locales in the system and check the success count */
 echo "*** Testing setlocale() : usage variations ***\n";
 
-function good_locale($locale) {
-	return $locale !== 'tt_RU@iqtelif.UTF-8';
+function good_locale($locale) { 
+    /**
+    * Note: no_NO is a bogus locale and should not be used, see https://bugzilla.redhat.com/show_bug.cgi?id=532487
+    **/
+	return $locale !== 'tt_RU@iqtelif.UTF-8' && substr($locale, 0, 5) !== "no_NO";
 }
 
 /* Prototype  : array list_system_locales( void )
From 3e6f17673338034724c8b637c214b53a51a9db14 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Wed, 1 Oct 2014 10:14:14 +0200
Subject: [PATCH] only no_NO.ISO-8859-1 have to be ignored

---
 ext/standard/tests/strings/setlocale_variation2.phpt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/standard/tests/strings/setlocale_variation2.phpt b/ext/standard/tests/strings/setlocale_variation2.phpt
index 10ae22f..5ebdfe8 100644
--- a/ext/standard/tests/strings/setlocale_variation2.phpt
+++ b/ext/standard/tests/strings/setlocale_variation2.phpt
@@ -20,9 +20,9 @@ echo "*** Testing setlocale() : usage variations ***\n";
 
 function good_locale($locale) { 
     /**
-    * Note: no_NO is a bogus locale and should not be used, see https://bugzilla.redhat.com/show_bug.cgi?id=532487
+    * Note: no_NO is a bogus locale and should not be used, see https://bugzilla.redhat.com/971416
     **/
-	return $locale !== 'tt_RU@iqtelif.UTF-8' && substr($locale, 0, 5) !== "no_NO";
+	return $locale !== 'tt_RU@iqtelif.UTF-8' && $locale !== 'no_NO.ISO-8859-1';
 }
 
 /* Prototype  : array list_system_locales( void )
